opencog/nlp/relex2logic/rules/gender.scm

```diff
@@ -0,0 +1,69 @@
+;check the atoms that should be grounded 
+(define (check-gender gtype)
+    (if (or (string=? (cog-name gtype) "masculine") (string=? (cog-name gtype) "feminine"))
+        (begin (stv 1 1))
+        (begin (stv 0 1))
+    )
+) 
+
+(define gender
+    (BindLink
+        (VariableList
+            (TypedVariableLink
+                (VariableNode "$a-parse")
+                (TypeNode "ParseNode")
+            )
+            (TypedVariableLink
+                (VariableNode "$word")
+                (TypeNode "WordInstanceNode")
+            )    
+            (TypedVariableLink
+                (VariableNode "$gtype")
+                (TypeNode "DefinedLinguisticConceptNode")
+            )
+        )
+        (AndLink
+            (WordInstanceLink
+                (VariableNode "$word")
+                (VariableNode "$a-parse")
+            )
+            (InheritanceLink
+                (VariableNode "$word")
+                (VariableNode "$gtype")
+                
+            )
+            (InheritanceLink
+                (VariableNode "$word")
+                (DefinedLinguisticConceptNode "person")
+                
+            )
+        (EvaluationLink
+                (GroundedPredicateNode "scm: check-gender")
+                (ListLink
+                    (VariableNode "$gtype")
+                )
+            )

```

Comment:
yes I think that is the same, and is a bit easier to read (and would run faster)